frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseOverrideForkPointAction.java

```diff
@@ -0,0 +1,144 @@
+import git4idea.config.GitConfigUtil;
+import io.vavr.collection.List;
+import lombok.CustomLog;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.i18nformatter.qual.I18nFormat;
+
+import com.virtuslab.gitmachete.backend.api.IGitMacheteBranch;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteCommit;
+import com.virtuslab.gitmachete.backend.api.SyncToParentStatus;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.OverrideForkPointDialog;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeyGitMacheteRepository;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeyProject;
+import com.virtuslab.logger.IEnhancedLambdaLogger;
+
+@CustomLog
+public abstract class BaseOverrideForkPointAction extends BaseGitMacheteRepositoryReadyAction
+    implements
+      IExpectsKeyProject,
+      IExpectsKeyGitMacheteRepository,
+      ISyncToParentStatusDependentAction {
+
+  @Override
+  public IEnhancedLambdaLogger log() {
+    return LOG;
+  }
+
+  @Override
+  public @I18nFormat({}) String getActionName() {
+    return getString("action.GitMachete.BaseOverrideForkPointAction.action-name");
+  }
+
+  @Override
+  public @I18nFormat({}) String getDescriptionActionName() {
+    return getString("action.GitMachete.BaseOverrideForkPointAction.description-action-name");
+  }
+
+  @Override
+  public @I18nFormat({GENERAL, GENERAL}) String getEnabledDescriptionFormat() {
+    return getString("action.GitMachete.BaseOverrideForkPointAction.description");
+  }
+
+  @Override
+  public String getCurrentBranchText() {
+    return getString("action.GitMachete.BaseOverrideForkPointAction.text.current-branch");
+  }
+
+  @Override
+  public List<SyncToParentStatus> getEligibleStatuses() {
+    return List.of(SyncToParentStatus.InSyncButForkPointOff);
+  }
+
+  @Override
+  @UIEffect
+  public void onUpdate(AnActionEvent anActionEvent) {
+    super.onUpdate(anActionEvent);
+    syncToParentStatusDependentActionUpdate(anActionEvent);
+  }
+
+  @Override
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    var project = getProject(anActionEvent);
+    var selectedVcsRepository = getSelectedGitRepository(anActionEvent).getOrNull();
+    var branchUnderAction = getNameOfBranchUnderAction(anActionEvent);
+    var branch = branchUnderAction.flatMap(pn -> getGitMacheteBranchByName(anActionEvent, pn)).getOrNull();
+
+    if (selectedVcsRepository == null || branch == null || branch.isRootBranch()) {
+      return;
+    }
+
+    var nonRootBranch = branch.asNonRootBranch();
+    var selectedCommit = new OverrideForkPointDialog(project, nonRootBranch.getParentBranch(), nonRootBranch)
+        .showAndGetSelectedCommit();
+    if (selectedCommit == null) {
+      log().debug(
+          "Selected commit to be the new fork point is null: most likely the action has been canceled from override-fork-point dialog");
```

Comment:
Nit, sounds more natural:
```suggestion
          "Commit selected to be the new fork point is null: most likely the action has been canceled from override-fork-point dialog");
```